File: app/src/main/java/com/oshamahue/kotlindemo/conversion/Conversion2ConditionAssignment.java
package com.oshamahue.kotlindemo.conversion;

import java.io.FileReader;
import java.io.IOException;

public class Conversion2ConditionAssignment {

    void whileAssignment(FileReader fr) throws IOException {
        int i;
        while ((i = fr.read()) != -1) {
            System.out.print((char) i);
        }

    }

}

File: app/src/main/java/com/oshamahue/kotlindemo/conversion/Conversion2ConditionAssignment.kt
package com.oshamahue.kotlindemo.conversion

import java.io.FileReader
import java.io.IOException

class Conversion2ConditionAssignment {

    @Throws(IOException::class)
    internal fun whileAssignment(fr: FileReader) {
        var i: Int =fr.read()
        while (i != -1) {
            print(i.toChar())
            i = fr.read()
        }

    }

}

